### PR TITLE
Add PipelineRun gc command

### PR DESCRIPTION
**Pull request by LinuxSuRen**

### CODE REVIEW

File: kubectl-plugin/pipeline/gc.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+func (o *gcOption) runE(cmd *cobra.Command, args []string) error {
+	cmd.Printf("start to gc PipelineRuns in %d namespaces\n", len(o.namespaces))
+	errorsNs := []string{}
+
+	for i := range o.namespaces {
+		ns := o.namespaces[i]
+		if err := o.cleanPipelineRunInNamespace(ns); err != nil {
+			cmd.PrintErrf("failed to clean PipelineRuns in '%s'\n", ns)
+		}
```
[the remark this answers is not in the document]
Comment (LinuxSuRen) [reply kubectl-plugin/pipeline/gc.go:109]: See https://github.com/kubesphere-sigs/ks/pull/179/files#diff-8372315828ae3452d68a065bb9a1c296af880b1c30c34133eacb3398aa47f40aR84

File: kubectl-plugin/pipeline/gc.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		if completionTime, parseErr := time.Parse(time.RFC3339, completionTimeStr); parseErr == nil {
+			return completionTime.Add(maxAge).Before(time.Now())
+		}
+	}
+	return false
+}
+
+func (o *gcOption) runE(cmd *cobra.Command, args []string) error {
+	cmd.Printf("start to gc PipelineRuns in %d namespaces\n", len(o.namespaces))
+	errorsNs := []string{}
```
[the remark this answers is not in the document]
Comment (LinuxSuRen) [reply kubectl-plugin/pipeline/gc.go:103]: Nice catch! https://github.com/kubesphere-sigs/ks/pull/179/commits/9749b9419edb239eceb6e91864abc24c2ae98558

File: kubectl-plugin/pipeline/gc.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+		ns := o.namespaces[i]
+		if err := o.cleanPipelineRunInNamespace(ns); err != nil {
+			cmd.PrintErrf("failed to clean PipelineRuns in '%s'\n", ns)
+		}
+	}
+
+	if len(errorsNs) > 0 {
+		return fmt.Errorf("gc failed in %d namespaces: %v", len(errorsNs), errorsNs)
+	}
+	return nil
```
[the remark this answers is not in the document]
Comment (JohnNiang) [reply kubectl-plugin/pipeline/gc.go:116]: When there is no need to clean PipelineRuns, the tip is as follow:

```go
starting to gc PipelineRuns in 2 namespaces
```

When I saw the tip, I thought it was still running if there was no `EOF` tip.

[the remark this answers is not in the document]
Comment (LinuxSuRen) [reply kubectl-plugin/pipeline/gc.go:117]: I've commented this log output. It is better to have a logger output to replace it.

See also https://github.com/kubesphere-sigs/ks/pull/179/commits/d46237c153a38c4c52acab964966f088d7b9eb62

File: kubectl-plugin/pipeline/gc.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	cmd = &cobra.Command{
+		Use:     "gc",
+		Short:   "Garbage collector for PipelineRuns",
+		Long:    "Clean all those old PipelineRuns by age or count",
+		PreRunE: opt.preRunE,
+		RunE:    opt.runE,
+	}
+
+	flags := cmd.Flags()
+	flags.IntVarP(&opt.maxCount, "max-count", "", 10,
```
Comment (cndoit18) [inline kubectl-plugin/pipeline/gc.go:30]: Add the `--verbose` parameter to provide more output information, see https://github.com/kubesphere-sigs/ks/pull/179/files#r697239943

link: http://www.linfo.org/rule_of_silence.html

Comment (LinuxSuRen) [reply kubectl-plugin/pipeline/gc.go:30]: A single option `--verbose` might not be enough. I prefer to support something like `-v=7` or a logger level. IMO, this is not part of this PR. And this is not a blocker for this feature. I will improve it in the further PRs.

I've created an issue to track the rules for code review. See https://github.com/kubesphere-sigs/ks/issues/185

Thanks for all your suggestions @cndoit18 and @JohnNiang

**Pull request merged**